src/deepsparse/server/build_logger.py

```diff
@@ -137,16 +136,51 @@ def build_function_loggers(
     return function_loggers
 
 
-def _create_system_logger(loggers: Dict[str, BaseLogger]) -> FunctionLogger:
-    # returns a function logger that matches to all system logs, logging
-    # every system call to each leaf logger
-    return _build_function_logger(
-        metric_function_cfg=MetricFunctionConfig(
-            func="identity", frequency=1, target_loggers=None
-        ),
-        target_identifier=f"category:{MetricCategories.SYSTEM.value}",
-        loggers=loggers,
-    )
+def create_system_loggers(
+    loggers: Dict[str, BaseLogger],
+    system_logging_config: Dict[str, Any],
+    available_system_groups: List[str] = AVAILABLE_SYSTEM_GROUPS,
+) -> List[Optional[FunctionLogger]]:
+    """
+    Create a system loggers (a list of FunctionLogger responsible for logging system metrics)
+    according to the configuration.
+
+    :param loggers: The created "leaf" loggers
+    :param system_logging_config: The system logging configuration
+    :return: A list of FunctionLogger instances
+    """
+    system_logging_enabled = system_logging_config.pop("enable", True)
+    if not system_logging_enabled:
+        return []
+    system_loggers = []
+    # extract the global `target_loggers` if present
+    target_loggers = system_logging_config.pop("target_loggers", None)
+
+    # iterate over the system group targets and create a system logger for each
+    for target_name, target_args in system_logging_config.items():
+        target_name = target_name.replace("_group", "")
+        # check for valid target name
+        if target_name not in available_system_groups:
+            raise ValueError(
+                f"Unknown system logging group: {target_name}. Expected one of: {available_system_groups}"
+            )
+        target_logging_enabled = system_logging_config.get("enable", True)
+        if not target_logging_enabled:
+            continue
+
+        system_loggers.append(
+            _build_function_logger(
+                metric_function_cfg=MetricFunctionConfig(
```

Comment:
why do we need a function logger here? what does the logger hierarchy for these system logs look like now?